torch/csrc/distributed/c10d/ProcessGroupNCCL.cpp

```diff
@@ -711,14 +718,25 @@ void ProcessGroupNCCL::WorkNCCL::synchronize() {
   }
 }
 
+void ProcessGroupNCCL::WorkNCCL::stashTensors(
+    std::vector<at::Tensor>& tensors) {
+  std::lock_guard<std::mutex> lock(stashMutex_);
+  stashed_for_allocator_safety_->insert(
```

Comment:
Do we ever expect to stash tensors more than once (e.g., do multiple inserts), or realistically does this only happen once (so `stashTensors` could take ownership of the vector directly)?